.github/workflows/ci.yml

```diff
@@ -0,0 +1,48 @@
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+name: ci
```

Comment:
Nit: Personally, I'd rename it to `test` or something along those lines. What if we add additional workflows down the line for, say, build a static library (hypothetical) or publishing the crate. Seems awkward to have one "ci" (noun) workflow and then others called, say, `publish` (verb). But not a biggie and personal preference, of course.

Reply:
I think 'ci' is fine for now and just an umbrella for all tasks. The checks show up like `ci / fmt (pull_request)` so we still retain the names of the individual tasks.

We can revisit if it gets confusing.